mlflow/models/evaluation/utils/trace.py

```diff
@@ -0,0 +1,117 @@
+import contextlib
+import inspect
+import logging
+from typing import Any
+
+import mlflow
+from mlflow.ml_package_versions import FLAVOR_TO_MODULE_NAME
+from mlflow.utils.autologging_utils import AUTOLOGGING_INTEGRATIONS, autologging_conf_lock
+
+_logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+@autologging_conf_lock
+def configure_autologging_for_evaluation(enable_tracing: bool = True):
+    """
+    Temporarily override the autologging configuration for all flavors during the model evaluation.
+    For example, model auto-logging must be disabled during the evaluation. After the evaluation
+    is done, the original autologging configurations are restored.
+
+    Args:
+        enable_tracing (bool): Whether to enable tracing for the supported flavors during eval.
+    """
+    flavor_to_original_config = {}
+    trace_enabled_flavors = []
+
+    # AUTOLOGGING_INTEGRATIONS can change during we iterate over flavors and enable/disable
+    # autologging, therefore, we snapshot the current configuration to restore it later.
+    global_config_snapshot = AUTOLOGGING_INTEGRATIONS.copy()
+
+    for flavor in FLAVOR_TO_MODULE_NAME:
+        # TODO: Remove this once Gluon autologging is actually removed
+        if flavor == "gluon":
+            continue
+
+        try:
+            if autolog := _get_autolog_function(flavor):
+                original_config = global_config_snapshot.get(flavor, {}).copy()
+
+                # If autologging is explicitly disabled, do nothing.
+                if original_config.get("disable", False):
+                    continue
+
+                elif enable_tracing and _is_trace_autologging_supported(flavor):
+                    # set all log_xyz params to False except log_traces
+                    new_config = {
+                        k: False if k.startswith("log_") else v for k, v in original_config.items()
+                    }
+                    new_config |= {"log_traces": True, "silent": True}
+                    _kwargs_safe_invoke(autolog, new_config)
+                    trace_enabled_flavors.append(flavor)
+                elif flavor in AUTOLOGGING_INTEGRATIONS:
+                    # For flavors that does not support tracing, disable autologging
+                    autolog(disable=True)
+
+                flavor_to_original_config[flavor] = original_config
+
+        except Exception as e:
+            if isinstance(e, ImportError):
+                _logger.debug(f"Failed to import {flavor}. Skip updating autologging.")
```

Comment:
```suggestion
                _logger.debug(f"{e!r}. Skip updating autologging.")
```

Can we include the original error message?